src/ILVerification/src/Verifier.cs

```diff
@@ -23,15 +23,21 @@ public class Verifier
             new Lazy<ResourceManager>(() => new ResourceManager("FxResources.ILVerification.SR", typeof(Verifier).GetTypeInfo().Assembly));
 
         private ILVerifyTypeSystemContext _typeSystemContext;
+        private VerifierOptions _verifierOptions;
 
-        public Verifier(IResolver resolver)
+        public Verifier(IResolver resolver) : this(resolver, GetDefaultVerifierOptions()) { }
+
+        public Verifier(IResolver resolver, VerifierOptions verifierOptions) : this(new ILVerifyTypeSystemContext(resolver), verifierOptions) { }
+
+        internal Verifier(ILVerifyTypeSystemContext context, VerifierOptions verifierOptions)
         {
-            _typeSystemContext = new ILVerifyTypeSystemContext(resolver);
+            _typeSystemContext = context;
+            _verifierOptions = verifierOptions ?? GetDefaultVerifierOptions();
```

Comment:
And this can be just `_verifierOptions = verifierOptions ?? new VerifierOptions();`. I do not think the GetDefaultVerifierOptions adds much value.

Reply:
(E.g. here is similar pattern in the framework: https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/src/System/Xml/Core/XmlReader.cs#L1767)

Reply:
Ok it was for future use however for now makes sense.